spmd/tensor/experimental_ops.py

```diff
@@ -16,33 +17,76 @@
 
 @register_prop_rule("aten.native_layer_norm.default")
 def _prop_native_layer_norm(op_schema: OpSchema) -> OutputSharding:
-    # TODO: Assert placement is Replicate() since we need to insert
-    # allreduce calls for _Partial results.
     input, normalized_shape, weight, bias, eps = op_schema.args_schema
-    out, mean, rstd = input, weight, weight
-    dims_list = op_schema.args_schema[1]
     assert isinstance(input, DTensorSpec)
+    assert isinstance(weight, DTensorSpec)
+    assert isinstance(bias, DTensorSpec)
+    assert isinstance(normalized_shape, (tuple, list))
+    assert all(isinstance(p, Replicate) for p in weight.placements)
+    assert all(isinstance(p, Replicate) for p in bias.placements)
+    # only the left-most (non-normalized) dimensions of the input can be sharded
+    batch_ndim = len(input.shape) - len(normalized_shape)
+    assert all(
+        isinstance(p, Replicate)
+        or (isinstance(p, Shard) and p.dim < batch_ndim,)
+        for p in input.placements
+    )
+    stats_spec = DTensorSpec(
+        mesh=weight.mesh,
+        placements=input.placements,
+        shape=torch.Size(
+            input.shape[:batch_ndim] + (1,) * len(normalized_shape)
+        ),
+        ndim=input.ndim,
+    )
+    return OutputSharding(output_spec=(input, stats_spec, stats_spec))
+
+
+@register_prop_rule("aten.native_layer_norm_backward.default")
+def _prop_native_layer_norm_backward(op_schema: OpSchema) -> OutputSharding:
+    (
+        grad,
+        input,
+        normalized_shape,
+        result1,
+        result2,
+        weight,
+        bias,
+        grad_input_mask,
+    ) = op_schema.args_schema
+    assert isinstance(grad, DTensorSpec)
+    assert isinstance(weight, DTensorSpec)
+    assert isinstance(bias, DTensorSpec)
+    assert isinstance(grad_input_mask, (list, tuple))
+    assert all(isinstance(s, Replicate) for s in weight.placements)
+    assert all(isinstance(s, Replicate) for s in bias.placements)
+    # ensure sharding on dim 0, which will trigger the "Partial" output on weight and bias grads
+    assert any(
+        isinstance(s, Shard) and s.dim == 0 for s in grad.placements
+    ), f"Got {grad.placements}"
+    weight_grad = DTensorSpec(
```

Comment:
could we add some comment about why grad sharding on dim 0 will produce partial gradients on weight and bias grad?